modules/bpi/bpi.module

```diff
@@ -397,6 +412,116 @@ function bpi_form_node_form_alter(&$form, &$form_state, $form_id) {
 }
 
 /**
+ * Decide if the given field is a container for the given bpi image type.
+ *
+ * @param string $field_name
+ *   The field name.
+ * @param array $node
+ *   The node.
+ * @param string $image_type
+ *   The bpi image type.
+ *
+ * @return bool
+ *   True if the field is a container for the the bpi image type.
+ */
+function bpi_is_bpi_image_field($field_name, array $node, $image_type) {
+  return bpi_get_image_type($field_name, $node) == $image_type;
+}
+
+/**
+ * Place syndicated images in node edit form.
+ *
+ * @param array $form
+ *   The form.
+ * @param string $current_language
+ *   The current language.
+ * @param array $syndicated_images
+ *   The syndicated images.
+ */
+function bpi_place_images(array &$form, $current_language, array $syndicated_images) {
+  $node = $form['#node'];
+  $image_field_names = bpi_fetch_image_fields($node->type);
+  if (!empty($image_field_names)) {
+    foreach ($syndicated_images as $image_type => $images) {
+      foreach ($image_field_names as $image_field_name) {
+        if (bpi_is_bpi_image_field($image_field_name, $node, $image_type)) {
+          $field = &$form[$image_field_name][$current_language];
+          $cardinality = isset($field['#cardinality']) ? $field['#cardinality'] : 1;
+          $number_of_images = count($images);
+          for ($i = 0; $i < min($cardinality, $number_of_images); $i++) {
+            if (is_object($images[$i]['@managed_file'])) {
+              $field[$i]['#default_value'] = array('fid' => $images[$i]['@managed_file']->fid);
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Replace inline images in content with references to syndicated images.
+ *
+ * @param string $field_name
+ *   The field name.
+ * @param string $content
+ *   The content.
+ * @param string $current_language
+ *    The current language.
+ * @param array $syndicated_images
+ *   The syndicated images.
+ *
+ * @return string
+ *   The content with inline images replaced by references to syndicated images.
+ */
+function bpi_place_inline_images($field_name, $content, $current_language, array $syndicated_images) {
+  if (!empty($syndicated_images)) {
+    $url_to_image_map = array();
+    foreach ($syndicated_images as $type => $images) {
+      foreach ($images as $image) {
+        $url_to_image_map[$image['path']] = $image;
+      }
+    }
+
+    // Find <img/> elements in content and replace with file placeholders.
+    $image_pattern = '@<img\s*[^>]*\ssrc=[\'"](?<url>[^\'">]+)[\'"][^>]*>@';
+
+    $content = preg_replace_callback($image_pattern, function($matches) use ($current_language, $url_to_image_map) {
+      $url = $matches['url'];
+      if (isset($url_to_image_map[$url])) {
+        $image = $url_to_image_map[$url];
+        $file = $image['@managed_file'];
+
+        // Build data for file placeholder.
```

Comment:
This format is directly coupled to the `media_wysiwyg` filter. It would be good to have this dependency described somewhere. I could not find any method in `media_wysiwyg` which we could at least add a comment stating this.